dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1045,20 +1044,9 @@ public Workflow getPublishedWorkflowByPath(@ApiParam(value = "repository path",
     public String verifiedSources(@ApiParam(value = "Workflow id", required = true) @PathParam("workflowId") Long workflowId) {
         Workflow workflow = workflowDAO.findById(workflowId);
         checkEntry(workflow);
-
-        Set<String> verifiedSourcesArray = new HashSet<>();
-        workflow.getWorkflowVersions().stream().filter(Version::isVerified)
-            .forEach((WorkflowVersion v) -> verifiedSourcesArray.add(v.getVerifiedSource()));
-
-        JSONArray jsonArray;
-        try {
-            jsonArray = new JSONArray(verifiedSourcesArray.toArray());
-        } catch (JSONException ex) {
-            throw new CustomWebApplicationException("There was an error converting the array of verified sources to a JSON array.",
-                HttpStatus.SC_INTERNAL_SERVER_ERROR);
-        }
-
-        return jsonArray.toString();
+        Set<String> verifiedSources = VerificationHelper.getVerifiedSources(workflow.getWorkflowVersions());
```

Comment:
Ditto, looks like this iterates through all source files for all versions on every call